SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc

```diff
@@ -0,0 +1,452 @@
+      std::vector<DigitizerUtility::EnergyDepositUnit> ionization_points;
+
+      // For each sim hit, super-charges (electron-holes) are created every 10um
+      primary_ionization(*it, ionization_points);
+      // Drift the super-charges (only electrons) to the collecting electrodes
+      const auto collection_points = drift(*it, pix3Ddet, bfield, ionization_points, true);
+
+      // compute induced signal on readout elements and add to _signal
+      // *ihit needed only for SimHit<-->Digi link
+      induce_signal(*it, simHitGlobalIndex, tofBin, pix3Ddet, collection_points);
+    }
+  }
+}
+
+const bool Pixel3DDigitizerAlgorithm::_is_inside_n_column(const LocalPoint& p) const {
+  return (p.perp() <= _np_column_radius);
+}
+
+const bool Pixel3DDigitizerAlgorithm::_is_inside_ohmic_column(const LocalPoint& p,
+                                                              const std::pair<float, float>& half_pitch) const {
+  // The four corners of the cell
+  return ((p - LocalVector(half_pitch.first, half_pitch.second, 0)).perp() <= _ohm_column_radius) ||
+         ((p - LocalVector(-half_pitch.first, half_pitch.second, 0)).perp() <= _ohm_column_radius) ||
+         ((p - LocalVector(half_pitch.first, -half_pitch.second, 0)).perp() <= _ohm_column_radius) ||
+         ((p - LocalVector(-half_pitch.first, -half_pitch.second, 0)).perp() <= _ohm_column_radius);
+}
+
+// Diffusion algorithm: Probably not needed,
+// Assuming the position point is given in the reference system of the proxy
+// cell, centered at the n-column corner.
+// The algorithm assumes only 1-axis could produce the charge migration, this assumption
+// could be enough given that the p-columns (5 um radius) are in the corners of the cell
+// (no producing charge in there)
+// The output is vector of newly created charge in the neighbour pixel i+1 or i-1,
+// defined by its position higher than abs(half_pitch) and the the sign providing
+// the addition or subtraction in the pixel  (i+-1)
+std::vector<DigitizerUtility::EnergyDepositUnit> Pixel3DDigitizerAlgorithm::diffusion(
+    const LocalPoint& pos,
+    const float& ncarriers,
+    const std::function<LocalVector(float, float)>& u_drift,
+    const std::pair<float, float> hpitches,
+    const float& thickness) {
+  // FIXME -- DM : Note that with a 0.3 will be enough (if using current sigma formulae)
+  //          With the current sigma, this value is dependent of the thickness,
+  //          Note that this formulae is coming from planar sensors, a similar
+  //          study with data will be needed to extract the sigma for 3D
+  const float _max_migration_radius = 0.4 * unit_um;
+  // Need to know which axis is the relevant one
+  int displ_ind = -1;
+  float pitch = 0.0;
+
+  // Check the group is near the edge of the pixel, so diffusion will
+  // be relevant in order to migrate between pixel cells
+  if (std::fabs(pos.x() - hpitches.first) < _max_migration_radius) {
```

Comment:
use `std::abs()` rather than `fabs()` (everywhere)

Reply:
It's changed, but is there any reason for that? I would say it is preferable not to use a overloaded variant to avoid potential behavior changes depending what includes are present (cmath, cstdlib, ..)

Reply:
It is preferable to use the modern `std::abs()` precisely because it is overloaded and will automatically choose the right behavior for each type. The old (pre-C++11) situation with `::abs()` and `fabs()` was suboptimal.

Reply:
ok, thanks for the clarification